### ISSUE TITLE
The auto-convert `.format()` to f-strings completely misses some `.format()` statements

### ISSUE THREAD

**Issue by john-science** [astral-sh/ruff#17972]
### Summary

In my [code](https://github.com/terrapower/armi/pull/2135/files#r2080518707), I had this format statement:

```python
" {0:^10s} {1:^25s}".format(str(total), str("Total Number of Warnings")
```

When I did this, `ruff` doesn't even register this f-string as existing:

```bash
ruff check --select=TID252 my_file.py
```

And when I did this, it fixed a dozen `.format()` statements in the file, but totally ignored this one (did not even mention it):

```bash
ruff check --select=TID252 . --unsafe-fixes --fix
```

I love `ruff`, by the way. You folks rock.

**Comment by ntBre** [comment]
Thanks for the kind words and the report!

This is the expected behavior for the exact command in the issue, which only selects [relative-imports (TID252)](https://docs.astral.sh/ruff/rules/relative-imports/#relative-imports-tid252) and not any rules related to `format`, as far as I know. However, I can also reproduce the issue in the [playground](https://play.ruff.rs/1b937575-33b2-4106-a305-e1f0b0f52a9c) when selecting [f-string (UP032)](https://docs.astral.sh/ruff/rules/f-string/#f-string-up032), which I assume you meant :)

I haven't identified the exact cause here yet, but it looks like the rule is not applied when one of the `format` arguments is a string literal. This triggers UP032, for example:

```python
warnings = "Total Number of Warnings"
" {0:^10s} {1:^25s}".format(str(total), str(warnings))
```

This is probably to avoid quoting issues, but maybe there's something more we can do here, especially on Python versions after 3.12 where this code is fine:

```python
f" {str(total):^10s} {str("Total Number of Warnings"):^25s}"
```

**Comment by MichaReiser** [comment]
See https://github.com/astral-sh/ruff/issues/17918 and https://github.com/astral-sh/ruff/issues/2031
